shared_buffer_pool.go

```diff
@@ -0,0 +1,58 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpc
+
+import "sync"
+
+// SharedBufferPool is a pool of buffers that can be shared.
+type SharedBufferPool interface {
+	Get(length int) []byte
+	Put(*[]byte)
+}
+
+// NewSimpleSharedBufferPool creates a new SimpleSharedBufferPool.
+func NewSimpleSharedBufferPool() SharedBufferPool {
+	return &simpleSharedBufferPool{
+		Pool: sync.Pool{
+			New: func() interface{} {
+				bs := make([]byte, 0)
+				return &bs
+			},
+		},
+	}
+}
+
+// simpleSharedBufferPool is a simple implementation of SharedBufferPool.
+type simpleSharedBufferPool struct {
+	sync.Pool
```

Comment:
Instead of a single `sync.Pool`, can we have the earlier implementation that you had over here. The one where you had four or five buckets of size in powers of 16. Thanks.

Reply:
Done with acec62662063acfd13ec30e74ebec877efadd777